libs/api/domains/national-registry-x-road/src/lib/utils/computeCountryResidence.ts

```diff
@@ -0,0 +1,59 @@
+import { NationalRegistryResidence } from '../../models/nationalRegistryResidence.model'
+import { DAY } from './constants'
+
+/**
+ * Compute a summary of how many days you've spent as a resident of each country in
+ * the last 365 days
+ *
+ * @param history list from national id registry of residences
+ * @returns map of country code to days spent in country in the last 365 days
+ */
+export const computeCountryResidence = (
```

Comment:
This function feels overly complex for a shared library and will be hard for other devs to contribute and understand.

This could be simplified a lot with an implementation similar to this:
```typescript
const last354days = xxx

return history.reduce((acc, item) =? {
  const numberOfDaysLivedInCountry = Math.max(0, differenceInDays(last365days, item.dateOfChange))
  return {
    ...acc,
    [item.country]: numberOfDaysLivedInCountry + (acc[item.country] || 0)
  }
}, {})
```

Reply:
@darrikonn I agree, that reads a lot better. Nut sure it captures the nuance of each case.. 

i agree about the premise of this being probably too involved to live in this library, in many cases.. I could move it to someplace else

Reply:
I think @darrikonn answer catches everything and if it doesn't then your tests should catch it. I would try it out.